modules/dxp/apps/scim/scim-client/src/main/java/com/liferay/scim/client/internal/configuration/SCIMClientOAuth2ApplicationConfigurationFactory.java

```diff
@@ -66,6 +73,19 @@ protected void activate(Map<String, Object> properties) throws Exception {
 				_oAuth2Application = _getOrAddOAuth2Application(
 					companyId, scimClientOAuth2ApplicationConfiguration);
 
+				if (_serviceRegistration == null) {
+					_serviceRegistration = bundleContext.registerService(
+						SCIMClientBearerTokenProvider.class,
+						new SCIMClientBearerTokenProvider(), null);
+				}
+
+				_serviceRegistration.setProperties(
+					HashMapDictionaryBuilder.<String, Object>put(
+						"clientId", _oAuth2Application.getClientId()
+					).put(
+						"companyId", companyId
+					).build());
+
```

Comment:
Because we are actually only allowing one SCIM client per company for the moment, we can simplify this code.

We can remove the `(_serviceRegistration == null)` check and add the properties directly into the registerService() invocation.